content/authentication/reading/add-service-dto/_index.md

```diff
@@ -301,4 +354,358 @@ Add this method to your `UserService` below the fields.
 ### Adding `EventService` and `EventCategoryService`
 
 We will add a service "layer" to our design that will be responsible for
-translating DTOs to Models and communication between the 
+translating DTOs to Models and communication between the `EventController` and
+`EventRepository`
+
+#### `EventService`
+
+Let's create another class in the `services` package named `EventService`.
+
+```java
+@Service
+public class EventService {
+
+    @Autowired
+    private EventRepository eventRepository;
+
+    @Autowired
+    private EventCategoryRepository categoryRepository;
+
+    @Autowired
+    private UserService userService;
+
+}
+```
+
+Our service will need references to the repositories so that it can access
+the database, and a reference to `UserService` so that it can retrieve the
+currently logged-in user.
+
+Next let's add some methods that will expose database functionality. Notice
+how we use the new `findAllByCreator` and `findByIdAndCreator` repository
+methods to filter events by user.
+
+```java
+    public List<Event> getAllEvents() {
+        return (List<Event>) eventRepository.findAll();
+    }
+
+    public List<Event> getAllEventsByCreator(User creator) {
+        return eventRepository.findAllByCreator(creator);
+    }
+
+    public Event getEventById(int id) {
+        return eventRepository.findById(id).orElseThrow(ResourceNotFoundException::new);
+    }
+
+    public Event getEventByIdAndCreator(int id, User creator) {
+        return eventRepository.findByIdAndCreator(id, creator).orElseThrow(ResourceNotFoundException::new);
+    }
+
+    public void removeEventById(int id) {
+        eventRepository.deleteById(id);
+    }
+```
+
+Last, we must add a `save` method, which
+takes in an `EventDTO` object and will translate it to our `Event` and
+`EventDetails` models before saving to the database.
+
+```java
+    public Event save(EventDTO eventDTO) {
+        Event event = new Event();
+        event.setName(eventDTO.getName());
+
+        EventDetails details = new EventDetails(eventDTO.getDescription(), eventDTO.getContactEmail());
+        event.setEventDetails(details);
+
+        event.setEventCategory(categoryRepository.findById(eventDTO.getCategoryId()).get());
```

Comment:
use .orElse(null) here, also consider using categoryService instead of Repository